app/models/ai_persona.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+class AiPersona < ActiveRecord::Base
+  class MultisiteHash
+    def initialize(id)
+      @hash = Hash.new({})
+      @id = id
+
+      MessageBus.subscribe(channel_name) { |message| @hash[message.data] = {} }
+    end
+
+    def channel_name
+      "/multisite-hash-#{@id}"
+    end
+
+    def current_db
+      RailsMultisite::ConnectionManagement.current_db
+    end
+
+    def [](key)
+      @hash.dig(current_db, key)
+    end
+
+    def []=(key, val)
+      @hash[current_db][key] = val
+    end
+
+    def flush!
+      @hash[current_db] = {}
+      MessageBus.publish(channel_name, current_db)
+    end
+  end
+
+  def self.persona_cache
+    @persona_cache ||= MultisiteHash.new("persona_cache")
+  end
+
+  def self.all_personas
+    persona_cache[:value] ||= AiPersona.all.map do |ai_persona|
```

Comment:
So this cache ends up having class definitions in it? :exploding_head: 

How does the output of `persona_cache[:value]` look?

Reply:
Ah I found it locally. With `Class.new` below, can you give it a name? Like `CustomPersona#{name}`? Otherwise the output looks like this and it's not immediately apparent which class is which:

```
AiPersona.all_personas.last
=> #<Class:0x00007f3753759f68>
```

Or maybe add a custom `.inspect` to the generated class to give different output?

Reply:
hmmm let me investigate

Reply:
`Object.const_set(name, ...)` wow never knew about this one, just need to be careful not to overlap here and to clear the const... not sure this is worth it

Reply:
inspect I guess is easiest here at least we don't need to worry about overlap

Reply:
hmm inspect does not give us much :( 
> Class.new { def inspect; "test"; end  }
=> #<Class:0x00007f52adabfac0>

I think we might as well live with this for now

.const_set hackery can work, but we need to be very careful to be namespaced perfectly, it can possibly lead to terribly complicated bugs

Reply:
Yeah even just inspect would be good, just helps with devex when debugging